linalg/x86_64/avx512/f32_store_clear.tmpliq

```diff
@@ -0,0 +1,83 @@
+{% comment %}
+Generate the code for the store and clear instructions.
+---
+Arguments:
+    mr - kernel size in number of elements
+    nr - kernel size in number of elements
+{% endcomment %}
+
+{%capture offset%}{% if msvc %} offset {%else%} rip + {%endif%}{%endcapture%}
+
+{% assign arch_mr = mr | divided_by:16 %}
+{% assign arch_mr_min_1 = mr | divided_by:16 | minus:1 %}
+{% assign nr_min_1 = nr | minus:1 %}
+
+{{L}}store:
+
+    mov     r10,    [rdi + 8]           // c ptr
+    mov     rsi,    [rdi + 16]          // row stride
+    mov     rbx,    [rdi + 24]          // col stride
+
+    // scatter operation
+
+    vpbroadcastd    zmm31,  esi
+    vpmulld         zmm31,  zmm31,  zmmword ptr [{{offset}} {{L}}numbers_seq_store]
+
+    // zmm31 is now a sequence of 0,esi,esi*2,esi*3...esi*15
+
+{% assign nr_min_1 = nr | minus:1 %}
+{% assign mr_arch = mr | divided_by:16 %}
+{% assign mr_arch_min_1 = mr | divided_by:16 | minus:1 %}
+
+    // r10 is cur col
+    imul    rsi,    16                  // stride for 16 elems
+
+{% for col in (0..nr_min_1) %}
+    mov     r9,     r10                 // cur row
+
+    {% for row in (0..mr_arch_min_1) %}
+        kxnorw      k1,k1,k1            // set writemask to ones
+        vscatterdps [r9 + zmm31]{k1}, zmm{{col | times:mr_arch | plus:row}}  
```

Comment:
I'm eyeing this line as a potential slowdown. vscatterdps is *hugely* expensive. Each row here is 43 uops!

Reply:
Woah, I did not know about that

Reply:
It's only 36 on Cannon Lake and Ice Lake, 43 on SKX; fwiw. https://www.agner.org/optimize/instruction_tables.pdf